lib/dpl/provider/code_deploy.rb

```diff
@@ -44,23 +44,22 @@ def revision
       end
 
       def revision_version_info
-        s3obj = s3api.head_object({
+        s3api.head_object({
           bucket: option(:bucket),
           key: s3_key
         })
+      rescue Aws::Errors::ServiceError
+        {}
       end
 
       def s3_revision
-        s3info = revision_version_info
         {
           revision_type: 'S3',
           s3_location: {
             bucket:      option(:bucket),
             bundle_type: bundle_type,
             key:         s3_key,
-            version:     s3info[:version_id],
-            e_tag:       s3info[:etag]
-          }
+          }.merge( revision_version_info )
```

Comment:
This will not work, and generate the error seen here: https://github.com/travis-ci/dpl/issues/741#issuecomment-360515841
The `version` and `e_tag` parameters have changed name, and there's a lot of other data coming from the s3 head_object call that you don't want.